Type, colour and spacing UI improvements

KenanYusuf opened the pull request:
The intention of these changes is to improve the usability of the devtools.

Changes include:
- Adding the `urql` brand purple to highlight interactive elements including active navigation items
- Adding `urql` logo with link to documentation
- Introduced a `<Toolbar />` component for sharing action style and logic between events and requests pages
- Reduced the amount of grey being used to separate sections of the UI
- Introduced `<Pane.Item />` and `<Pane.Header />` components to reduce style duplication across pages
- Tightened spacing across all parts of the UI, making each page and section consistent with one another
- Used dividing lines to separate pane items

Please check out the before/after screenshots below - happy to take any suggestions!

| Before | After |
| - | - |
| <img width="1159" alt="explorer-before" src="https://user-images.githubusercontent.com/9557798/116547195-801b0480-a8ea-11eb-8d9d-743341f1827d.png"> | <img width="1159" alt="explorer-after" src="https://user-images.githubusercontent.com/9557798/116546868-10a51500-a8ea-11eb-855d-c6006f83746e.png"> |
| <img width="1159" alt="events-before" src="https://user-images.githubusercontent.com/9557798/116547203-80b39b00-a8ea-11eb-9cdd-438b58154c6c.png"> | <img width="1159" alt="events-after" src="https://user-images.githubusercontent.com/9557798/116547206-814c3180-a8ea-11eb-8dc8-592a542b897d.png"> |
| <img width="1159" alt="request-before" src="https://user-images.githubusercontent.com/9557798/116547214-8315f500-a8ea-11eb-93c3-77b34e89d7d1.png"> | <img width="1159" alt="request-after" src="https://user-images.githubusercontent.com/9557798/116547209-81e4c800-a8ea-11eb-8c79-aa446d038a71.png"> |

_Apologies for the file change count... A lot of snapshots got updated 😬_

kadikraman commented:
Looking good! 🌟  Just need to update the header to work on dark mode 🕶️ 

![Screenshot 2021-04-29 at 15 44 17](https://user-images.githubusercontent.com/6534400/116570403-28889300-a902-11eb-95eb-d45cee6ef6bf.png)

andyrichardson commented:
> Looking good! 🌟 Just need to update the header to work on dark mode 🕶️
> 
> ![Screenshot 2021-04-29 at 15 44 17](https://user-images.githubusercontent.com/6534400/116570403-28889300-a902-11eb-95eb-d45cee6ef6bf.png)

Strange - I'm seeing this in the browser extension

![Screenshot 2021-04-29 at 15 50 40](https://user-images.githubusercontent.com/10779424/116571179-bf554f80-a902-11eb-9e94-3dbb8cb12e3d.png)

kadikraman commented:
> > Looking good! 🌟 Just need to update the header to work on dark mode 🕶️
> > ![Screenshot 2021-04-29 at 15 44 17](https://user-images.githubusercontent.com/6534400/116570403-28889300-a902-11eb-95eb-d45cee6ef6bf.png)
> 
> Strange - I'm seeing this in the browser extension
> 
> ![Screenshot 2021-04-29 at 15 50 40](https://user-images.githubusercontent.com/10779424/116571179-bf554f80-a902-11eb-9e94-3dbb8cb12e3d.png)

Oh, that is strange! Electron being awkward? 🤔

andyrichardson commented:
This is 🔥 🔥 🔥 

If you feel up to the task, I'd totally be open to us having two separate themes for light and dark.

There are a fair few places where color doesn't transition well when going from dark->light or the other way round.

E.g. The hover in dark mode I'd suggest we don't add such a saturated hue.

![Screenshot 2021-04-29 at 15 50 40](https://user-images.githubusercontent.com/10779424/116571495-093e3580-a903-11eb-9b8d-c02e14cd6314.png)

andyrichardson commented on src/panel/components/Navigation.tsx:
```diff
@@ -9,42 +10,66 @@ export const Navigation: FC<
   right: 0;
+`;
+
+const Item = styled.a<{ alignRight?: boolean }>`
```
Nice 👍

andyrichardson commented on src/panel/components/Toolbar.fixture.tsx:
```diff
@@ -0,0 +1,32 @@
+
+export default {
+  basic: (
+    <Toolbar
```
Thanks for adding a fixture!

andyrichardson reviewed (approved):
Looks dope!

I'd be tempted to keep existing composition (providing children) rather than props for most presentational elements so we don't lose any flexibility (additional props, changing elements, ref access, etc).

Not blocking though! Maybe others have thoughts too :+1:

wgolledge reviewed (approved):
Awesome 🔥 

This looks so much cleaner/more consistent

KenanYusuf commented on src/panel/App.css:
```diff
@@ -12,3 +12,13 @@ code,
   font-family: "Roboto Mono", monospace;
 }
+
+button {
```
Yeah, good point. My preference is always to reset on a global level to avoid bloating compnoent styles but I can see that the current approach is to unset on a per-component basis. If we went with a global reset for all HTML elements there'd be a good cleanup opportunity?

@andyrichardson any thoughts on this?

KenanYusuf commented on src/panel/components/Navigation.tsx:
```diff
@@ -9,42 +10,66 @@ export const Navigation: FC<
+const Logo = styled(Icon)`
+  width: 32px;
+  height: 19px;
+  > * {
```
Good spot - have fixed the other instance of this also https://github.com/FormidableLabs/urql-devtools/pull/355/commits/c339765ffae867a6707a71494c3ff50e914d9620

KenanYusuf commented:
Thank you all for your comments!

@kadikraman good spot on the navigation background - looks like that was only happening on the electron version which is why I missed it - will be sure to test on that next time! Fixed with this https://github.com/FormidableLabs/urql-devtools/pull/355/commits/46b9b7b820ca8eb8310f53d45fc5de2311ad024a

@andyrichardson agree on your point on the composition of `Pane.Item` - have removed the prop and added `Pane.ItemTitle` so that we're still benefiting from reusing those styles https://github.com/FormidableLabs/urql-devtools/pull/355/commits/742e098ba300479ba04842f208a237bd3d6fbd09

On the point of theming, as discussed I will look at that in a separate PR 😄

kadikraman reviewed (approved):
Tested on React Native and looks great!  🎉  😍
